### PR TITLE
apt-hook: fix cloud identification

### PR DESCRIPTION
**Pull request by orndorffgrant**

### CODE REVIEW

File: features/apt_messages.feature [hunk 1]
```diff
@@ -605,7 +605,8 @@ Feature: APT Messages
     Scenario Outline: AWS URLs
         Given a `<release>` machine with ubuntu-advantage-tools installed
         When I run `apt-get update` with sudo
-        When I run `pro refresh messages` with sudo
+        When I run `apt-get install ansible -y` with sudo
```
Comment (lucasmoura) [inline features/apt_messages.feature:608]: Just double checking, why are we installing ansible here ?

Comment (orndorffgrant) [reply features/apt_messages.feature:608]: For the bionic version of this test, there needs to be a package with an update available in esm-apps to get the message

**Review by renanrodrigo — APPROVED** [review]
Given this PR is about clouds and hooks, maybe an extra commit may fit in:
We have this warning when building that can be removed I guess?
```
json-hook.cc: In function ‘ESMContext get_esm_context()’:
json-hook.cc:242:10: warning: unused variable ‘non_azure_cloud’ [-Wunused-variable]
  242 |     bool non_azure_cloud = cloud_id != NONE && cloud_id != AZURE;
      |          ^~~~~~~~~~~~~~~
```

**Comment by orndorffgrant** [comment]
@renanrodrigo I totally missed that, yes I'll fix it as an extra commit in this PR

**Comment by orndorffgrant** [comment]
rebased and updated - hopefully that'll help CI as well

**Comment by orndorffgrant** [comment]
CI fails are unrelated

**Pull request merged**